portfolio/src/main/webapp/script.js

```diff
@@ -33,76 +33,118 @@ function addRandomFact() {
+    headingBgColor: 'rgba(173,255,47,.2)',
+  },
+];
 
 /**
- * Advances the background image by a given offset from the current image 
+ * Advances the background image by a given offset from the current image.
  */
 function changeBackground(offset) {
-  // List of background images and their color themes
-  const backgroundImgs = [
-    {
-      img: 'forest.jpg',
-      headingColor: '#523029',
-      headingBgColor: 'rgba(160,82,45,.1)',
-    }, {
-      img: 'ocean.jpg',
-      headingColor: 'navy',
-      headingBgColor: 'rgba(65,105,225,.1)',
-    }, {
-      img: 'mountain.jpg',
-      headingColor: 'midnightblue',
-      headingBgColor: 'rgba(0,128,128,.15)',
-    }, {
-      img: 'balloons.jpg',
-      headingColor: 'indianred',
-      headingBgColor: 'rgba(255,215,0,.2)',
-    }, {
-      img: 'iceberg.jpg',
-      headingColor: 'darkslategray',
-      headingBgColor: 'rgba(72,61,139,.2)',
-    }, {
-      img: 'grass.jpg',
-      headingColor: 'green',
-      headingBgColor: 'rgba(173,255,47,.2)',
-    },
-  ];
-
-  // Update the index of the current background image
-  curBackground = (curBackground + offset) % backgroundImgs.length;
+  // Update the index of the current background image.
+  currentBackgroundIndex = (currentBackgroundIndex + offset) % backgroundThemes.length;
 
-  // Wrap around to the last image if the index is negative
-  if (curBackground < 0) {
-    curBackground += backgroundImgs.length;
+  // Wrap around to the last image if the index is negative.
+  if (currentBackgroundIndex < 0) {
+    currentBackgroundIndex += backgroundThemes.length;
   }
 
-  const newBackground = backgroundImgs[curBackground];
+  const newBackground = backgroundThemes[currentBackgroundIndex];
 
-  // Update the page with the new background image
+  // Update the page with the new background image.
   const html = document.documentElement;
   html.style.backgroundImage = `url(images/${newBackground.img})`;
 
-  // Update the headings to the color theme for the new image
+  // Update the headings to the color theme for the new image.
   updateColorTheme(newBackground.headingColor, newBackground.headingBgColor);
 }
 
 /**
- * Updates the heading color CSS properties
+ * Updates the heading color CSS properties.
  */
 function updateColorTheme(headingColor, headingBgColor) {
   document.body.style.setProperty('--heading-color', headingColor);
   document.body.style.setProperty('--heading-bg-color', headingBgColor);
+}
+
+/**
+ * Fetches the list of comments from the server and adds them to the DOM.
+ */
+async function getComments() {
+  const maxComments = document.getElementById('max-comments').value;
+  const response = await fetch(`/data?max-comments=${maxComments}`);
+  const comments = await response.json();
+
+  const commentsContainer = document.getElementById('comments-container');
+  commentsContainer.innerHTML = '';
+
+  // Create <h4> and <p> elements for each comment's author and text.
+  for (const comment of comments) {
```

Comment:
Just want to mention that this could also be done with a stream, but what you have is fine as well :) :

`comments.forEach(comment => {`